Sources/LanguageServerProtocol/Hover.swift

```diff
@@ -77,3 +93,51 @@ extension HoverResponse: Codable {
     try container.encodeIfPresent(range.map { PositionRange(wrappedValue: $0) }, forKey: .range)
   }
 }
+
+extension MarkedString: Codable {
+  public init(from decoder: Decoder) throws {
+    if let value = try? decoder.singleValueContainer().decode(String.self) {
+      self = .markdown(value: value)
+    } else if let codeBlock = try? MarkdownCodeBlock(from: decoder) {
+      self = .codeBlock(language: codeBlock.language, value: codeBlock.value)
+    } else {
+      let context = DecodingError.Context(codingPath: decoder.codingPath, debugDescription: "MarkedString is neither pure string nor code block")
+      throw DecodingError.dataCorrupted(context)
+    }
+  }
+
+  public func encode(to encoder: Encoder) throws {
+    switch self {
+    case .markdown(value: let value):
+      try value.encode(to: encoder)
+    case .codeBlock(language: let language, value: let value):
+      try MarkdownCodeBlock(language: language, value: value).encode(to: encoder)
+    }
+  }
+}
+
+extension HoverResponseContents: Codable {
+  public init(from decoder: Decoder) throws {
+    if let value = try? MarkedString(from: decoder) {
+      self = .markedString(value)
+    } else if let value = try? [MarkedString](from: decoder) {
+      self = .markedStrings(value)
+    } else if let value = try? MarkupContent(from: decoder) {
```

Comment:
We should try decoding as `MarkupContent` first since it avoids attempting to deserialize the other formats in the common case.  The `MarkedString` response types are deprecated.

Reply:
Flipped the deserialization order